core/modules/redirect/redirect.admin.inc

```diff
@@ -687,29 +687,7 @@ function redirect_settings_form($form, &$form_state) {
     '#description' => t('Only redirects managed by the redirect module itself will be deleted. Redirects managed by other modules will be left alone.'),
   );
 
-  $form['actions']['#type'] = 'actions';
-  $form['actions']['submit'] = array(
-    '#type' => 'submit',
-    '#value' => t('Save configuration'),
-  );
-  $form['#submit'][] = 'redirect_settings_form_submit';
-  return $form;
-}
-
-/**
- * Form submit handler; clears the page cache.
- *
- * @see settings_form()
- */
-function redirect_settings_form_submit($form, &$form_state) {
-  $values = $form_state['values'];
-  $config = config('redirect.settings');
-  $config->set('auto_redirect', (bool) $values['auto_redirect']);
-  $config->set('passthrough_querystring', (bool) $values['passthrough_querystring']);
-  $config->set('additional_statuses', (bool) $values['additional_statuses']);
-  $config->set('purge_inactive', (bool) $values['purge_inactive']);
```

Comment:
The function should be able to detect that it's saving data from a `checkbox` (not`checkboxes`) element and cast it.